vendor/github.com/prometheus/prometheus/tsdb/postings_for_matchers_cache.go

```diff
@@ -241,7 +250,7 @@ func (c *PostingsForMatchersCache) postingsForMatchersPromise(ctx context.Contex
 
 	sizeBytes := int64(len(key) + size.Of(promise))
 
-	c.onPromiseExecutionDone(ctx, key, c.timeNow(), sizeBytes, promise.err)
+	c.onPromiseExecutionDone(ctx, key, ts, sizeBytes, promise.err)
```

Comment:
This subtle change could be quite impactful. It was done on purpose reading `c.timeNow()` at this point, to make the cache TTL lasting longer. The problem is that if you're running a very complex index lookup that takes few seconds, by the time you reach this point, the TTL of 10s may be either already expired or close to expire, effectively invalidating the benefits of the cache.

Reply:
Good catch, I'll revert this and find another way to do this.